central/graphql/resolvers/node_components_test.go

```diff
@@ -0,0 +1,480 @@
+
+	// node datastore
+	riskMock := mockRisks.NewMockDataStore(gomock.NewController(s.T()))
+	nodeStore := nodePostgres.CreateTableAndNewStore(s.ctx, s.T(), s.db, s.gormDB, false)
+	nodeIndexer := nodePostgres.NewIndexer(s.db)
+	nodeSearcher := nodeSearch.NewV2(nodeStore, nodeIndexer)
+	nodePostgresDataStore := nodeDackboxDataStore.NewWithPostgres(nodeStore, nodeIndexer, nodeSearcher, riskMock, ranking.NewRanker(), ranking.NewRanker())
+	nodeGlobalDatastore, err := nodeGlobalDataStore.New(nodePostgresDataStore)
+	s.NoError(err, "Failed to create nodeGlobalDatastore")
+	s.resolver.NodeGlobalDataStore = nodeGlobalDatastore
+
+	// nodeComponent datastore
+	nodeCompStore := nodeComponentPostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	nodeCompIndexer := nodeComponentPostgres.NewIndexer(s.db)
+	nodeCompSearcher := nodeComponentSearch.New(nodeCompStore, nodeCompIndexer)
+	s.resolver.NodeComponentDataStore = nodeComponentDataStore.New(nodeCompStore, nodeCompIndexer, nodeCompSearcher, riskMock, ranking.NewRanker())
+
+	// nodeComponentCVEEdge datastore
+	nodeComponentCveEdgeStore := nodeComponentCVEEdgePostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	nodeCompontCveEdgeIndexer := nodeComponentCVEEdgePostgres.NewIndexer(s.db)
+	nodeComponentCveEdgeSearcher := nodeComponentCVEEdgeSearch.New(nodeComponentCveEdgeStore, nodeCompontCveEdgeIndexer)
+	nodeComponentCveEdgeDatastore, err := nodeComponentCVEEdgeDataStore.New(nodeComponentCveEdgeStore, nodeCompontCveEdgeIndexer, nodeComponentCveEdgeSearcher)
+	s.NoError(err)
+	s.resolver.NodeComponentCVEEdgeDataStore = nodeComponentCveEdgeDatastore
+
+	// cluster datastore
+	clusterStore := clusterPostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	clusterHealthStore := clusterHealthPostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	clusterIndexer := clusterPostgres.NewIndexer(s.db)
+	connMgr := connection.ManagerSingleton()
+	clusterDatastore, err := clusterDataStore.New(clusterStore, clusterHealthStore, nil, nil, nil, nil, nil, nodeGlobalDatastore, nil, nil, nil, nil, nil, nil, nil, connMgr, nil, nil, ranking.NewRanker(), clusterIndexer, nil)
+	s.NoError(err)
+	s.resolver.ClusterDataStore = clusterDatastore
+
+	// Sac permissions
+	s.ctx = sac.WithAllAccess(s.ctx)
+
+	// loaders used by graphql layer
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.Node{}), func() interface{} {
+		return loaders.NewNodeLoader(nodePostgresDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.NodeComponent{}), func() interface{} {
+		return loaders.NewNodeComponentLoader(s.resolver.NodeComponentDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.NodeCVE{}), func() interface{} {
+		return loaders.NewNodeCVELoader(s.resolver.NodeCVEDataStore)
+	})
+	s.ctx = loaders.WithLoaderContext(s.ctx)
+
+	// Add test data to DataStores
+	testClusters, testNodes := testClustersWithNodes()
+	for _, cluster := range testClusters {
+		err = clusterDatastore.UpdateCluster(s.ctx, cluster)
```

Comment:
why `UpdateCluster` instead of `AddCluster`?

Reply:
AddCluster generates clusterID, which would be different from constant IDs used in tests.